mlflow/store/jobs/sqlalchemy_store.py

```diff
@@ -221,8 +227,25 @@ def list_jobs(
                     break
 
                 # Yield each job
-                for job in jobs:
-                    yield job.to_mlflow_entity()
+                if params:
+                    for job in jobs:
```

Comment:
I didn't mean to remove the function. I was suggesting we move it outside the loop.